mapactionpy_arcmap/MapChef.py

```diff
@@ -47,6 +46,9 @@ def removeLayers(self):
                 arcpy.mapping.RemoveLayer(df, lyr)
         self.mxd.save()
 
+    # APS 06/09/2019 The `cook()` method is much too long and has too many nested `if`s and `for`s. Please split this
+    # into the several separate private methods for handling the different scenarios, data types, error conditions 
+    # etc. This refactoring would benefit from adding some unittests surrounding it.
     def cook(self, productName, countryName):
```

Comment:
I've not fully tested it, but I believe there is a bug somewhere in the `cook()` method. If there are two or more files that match a regex for a particular layer, then there should be an error raised, warning added to the report, or some kind of user feedback. I don't think this happens and cannot see if or where this scenario is handled.